packages/lesswrong/server/repos/CommentsRepo.ts

```diff
@@ -75,6 +75,54 @@ export default class CommentsRepo extends AbstractRepo<DbComment> {
     `, [limit]);
   }
 
+  async getPopularPollComments (limit: number): Promise<(DbComment)[]> {
+    return await this.manyOrNone(`
+      SELECT *
+      FROM public."Comments" AS c
+      WHERE c."parentCommentId" = 'NtsPs9wcwrpeK6KYL'
+      ORDER BY c."baseScore" DESC
+      LIMIT $1
+    `, [limit]);
+  }
+
+  async getPopularPollCommentsWithUserVotes (userId:string, limit: number): Promise<(DbComment)[]> {
+    return await this.manyOrNone(`
+    SELECT *
+    FROM public."Comments" AS c
+    INNER JOIN public."Votes" AS v ON c._id = v."documentId"
+    WHERE
+      c."parentCommentId" = 'NtsPs9wcwrpeK6KYL'
+      AND v."userId" = $1
+      AND v."extendedVoteType"->'reacts'->0->>'vote' = 'created'
+      AND v.cancelled IS NOT TRUE
+      AND v."isUnvote" IS NOT TRUE
+    ORDER BY c."baseScore" DESC
+    LIMIT $2
+    `, [userId, limit]);
+  }
+
+  async getPopularPollCommentsWithTwoUserVotes (userId:string, targetUserId:string, limit: number): Promise<(DbComment)[]> {
+    return await this.manyOrNone(`
+      SELECT *
```

Comment:
I'd refactor this query to filter `votes` table before doing the self join to avoid duplicated all your where clauses. So use a WITH statement for something like votes_filtered, then self-join on that. Will be more readable.